static/script/devices/storage/cookie.js

```diff
@@ -25,107 +25,128 @@
+         * @param {Object} [opts]
+         * @param {String} [opts.domain] The domain value of the cookie, if not provided this is not set on the cookie
+         * @param {Boolean} [opts.isPathless] If <code>true</code> sets the path to '/' else retrieves the path from the location
+         */
+        var CookieStorage = StorageProvider.extend(/** @lends antie.devices.storage.CookieStorage.prototype */{
+            /**
+             * @constructor
+             * @ignore
+             */
+            init: function(namespace, opts) {
+                this._super();
 
-				this._namespace = namespace;
+                this._namespace = namespace;
+                this._opts = opts || {};
 
-				var cookie = readCookie(namespace);
+                var cookie = readCookie(namespace);
 
-				if(cookie) {
-					this._valueCache = Device.prototype.decodeJson(cookie);
-					if(this._valueCache) {
-						this._save();
-					} else {
-						this._valueCache = {};
-					}
-				}
-			},
-			setItem: function(key, value) {
-				this._super(key, value);
-				this._save();
-			},
-			removeItem: function(key) {
-				this._super(key);
-				this._save();
-			},
-			clear: function() {
-				this._super();
-				this._save();
+                if(cookie) {
+                    this._valueCache = Device.prototype.decodeJson(cookie);
+                    if(this._valueCache) {
+                        this._save();
+                    } else {
+                        this._valueCache = {};
+                    }
+                }
+            },
+            setItem: function(key, value) {
+                this._super(key, value);
+                this._save();
+            },
+            removeItem: function(key) {
+                this._super(key);
+                this._save();
+            },
+            clear: function() {
+                this._super();
+                this._save();
 
-				// delete it from the stored namespaces
-				// so it will be reloaded the next time
-				// we get it
-				delete namespaces[this._namespace];
-			},
-			_isEmpty: function() {
-				var prop;
-				for(prop in this._valueCache) {
-					return false;
-				}
-				return true;
-			},
-			_save: function() {
-				if(this._isEmpty()) {
-					eraseCookie(this._namespace);
-				} else {
-					var json = Device.prototype.encodeJson(this._valueCache);
-					createCookie(this._namespace, json);
-				}
-			}
-		});
-
-		Device.prototype.getPersistentStorage = function(namespace) {
-			if(!namespaces[namespace]) {
-				namespaces[namespace] = new CookieStorage(namespace);
-			}
-			return namespaces[namespace];
-		};
-	}
-);
+                // delete it from the stored namespaces
+                // so it will be reloaded the next time
+                // we get it
+                delete namespaces[this._namespace];
+            },
+            _isEmpty: function() {
+                var prop;
+                for(prop in this._valueCache) {
```

Comment:
This for...in without any hasOwnProperty makes me uncomfortable, but it was in the original code and I don't propose we change it now.